openpype/lib/build_template.py

```diff
@@ -0,0 +1,28 @@
+import avalon
+import importlib
+
+concrete_loaders_modules = {
+    'maya': 'openpype.hosts.maya.api.template_loader'
+}
+
+
+class BuildWorkfileTemplate:
+
+    def process(self):
+        dcc = avalon.io.Session['AVALON_APP']
+        module_path = concrete_loaders_modules.get(dcc, None)
+
+        if not module_path:
+            raise ValueError("Template not found for DCC '{}'".format(dcc))
+
+        module = importlib.import_module(module_path)
+        if not hasattr(module, 'TemplateLoader'):
```

Comment:
I would preffer to look for classes inheriting from `AbstractTemplateLoader` and `AbstractPlaceholder`. This will cause that all hosts will have same class names which is hell if you need to find out where is one specific class imported and used.

Reply:
Also would be better to look for attributes in `avalon.registered_host()` rather than predefine host specific path in dictionary.

Reply:
Hi,
Could you elaborate when you say "look for classes inheriting from AbstractTemplateLoader" 
Do you mean developing some sort of discover equivalent for template loaders on the project or locally on the file ?
Thanks